modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,519 @@
+      sirdataDomain = 'cookieless-data.com';
+      sendWithCredentials = false;
+      gdprApplies = false;
+    }
+  } else if (gdprApplies && gdprApplies !== '0' && gdprApplies !== 'null') {
+    if (userConsent && userConsent.gdpr && userConsent.gdpr.vendorData && userConsent.gdpr.vendorData.vendor && userConsent.gdpr.vendorData.vendor.consents) {
+      if (!userConsent.gdpr.vendorData.vendor.consents[53] || !userConsent.gdpr.vendorData.purpose.consents[1] || !userConsent.gdpr.vendorData.purpose.consents[3]) {
+        sirdataDomain = 'cookieless-data.com';
+        sendWithCredentials = false;
+      }
+    } else {
+      sirdataDomain = 'cookieless-data.com';
+      sendWithCredentials = false;
+    }
+  }
+
+  var actualUrl = null;
+  try {
+    actualUrl = window.top.location.href;
+  } catch (e) {}
+  if (!actualUrl && config.params.actualUrl) {
+    try {
+      actualUrl = new URL(config.params.actualUrl);
+    } catch (err) {}
+  }
+
+  const url = 'https://kvt.' + sirdataDomain + '/api/v1/public/p/' + config.params.partnerId + '/d/' + config.params.key + '/s?callback=&gdpr=' + gdprApplies + '&gdpr_consent=' + tcString + (actualUrl ? '&url=' + actualUrl : '');
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          if (data && data.segments) {
+            addSegmentData(adUnits, data, config, onDone, globalConfig);
+          } else {
+            onDone();
+          }
+        } catch (err) {
+          onDone();
+          utils.logError('unable to parse Sirdata data');
+        }
+      } else if (req.status === 204) {
+        onDone();
+      }
+    },
+    error: function () {
+      onDone();
+      utils.logError('unable to get Sirdata data');
+    }
+  },
+  null,
+  {
+    contentType: 'text/plain',
+    method: 'GET',
+    withCredentials: sendWithCredentials,
+    referrerPolicy: 'unsafe-url',
+    crossOrigin: true
+  });
+}
+
+export function setGlobalOrtb2(segments, categories, globalConfig) {
+  var ortb2Valid = true;
+
+  try {
+    if (parseFloat(globalConfig.version.substring(1)) < 4.3) {
+      ortb2Valid = false;
+    }
+  } catch (er) {}
+
+  if (!ortb2Valid) {
+    // do it with FPD
+    return setGlobalFpd(segments, categories, globalConfig);
+  }
+
+  try {
+    globalConfig.setConfig({
+      ortb2: {
+        site: {
+          ext: {
+            data: {
+              sd_rtd: categories
+            }
+          }
+        },
+        user: {
+          ext: {
+            data: {
+              sd_rtd: segments
+            }
+          }
+        }
+      }
+    });
+  } catch (err) {
+    utils.logError(err.message)
+  }
+
+  return true;
+}
+
+export function setGlobalFpd(segments, categories, globalConfig) {
+  try {
+    globalConfig.setConfig({
+      fpd: {
+        context: {
+          data: {
+            sd_rtd: categories
+          }
+        },
+        user: {
```

Comment:
site.user.data is special and should conform to the developing spec in #6057 did you mean site.user.ext.data? we've tried to be really verbose at the big example that is on https://docs.prebid.org/features/firstPartyData.html

Reply:
The module is designed for ORTB2 but automatically switch to FPD when when Prebid version is < 4.29, using the converion proposed on the page https://docs.prebid.org/features/firstPartyData.html

![image](https://user-images.githubusercontent.com/33549554/113359430-0bcf5e80-9348-11eb-8bc7-56b390a4ce8a.png)

Reply:
The module is designed for ORTB2 but automatically switch to FPD when when Prebid version is < 4.29, using the converion proposed on the page https://docs.prebid.org/features/firstPartyData.html

![image](https://user-images.githubusercontent.com/33549554/113359430-0bcf5e80-9348-11eb-8bc7-56b390a4ce8a.png)

Reply:
I am not sure this check is useful, as your code is not in 4.29 or before. I think it is safe for you to assume the publisher is using 4.32 plus. 

Also, your code check would seem to pass for 4.4, which is less than 4.30, because it isnt 4.04. I recommend you delete this check and assume everyone is passing this check. 

I see the distinction now though,  setBidderOrtb2 sets user.ext.data now user.datta

Reply:
ok, thank you